drivers/ukbus/pci/pci_bus.c

```diff
@@ -158,6 +167,272 @@ void _pci_register_driver(struct pci_driver *drv)
 	uk_list_add_tail(&drv->list, &ph.drv_list);
 }
 
+static int pci_get_bar(struct pci_device *dev, uint8_t idx, __paddr_t *bar_out,
+		       uint8_t *is_64bit)
+{
+	uint32_t cfg, offset, bar_low, bar_high;
+	__paddr_t base;
+
+	UK_ASSERT(dev);
+	UK_ASSERT(bar_out);
+	UK_ASSERT(idx < 6);
```

Comment:
In https://github.com/unikraft/unikraft/pull/1124/commits/171bf6bc4da8192c511746c061e08d6765f3655e I made a macro for maximum BAR; maybe it can be done in this PR?